docs/introduction/faq.md

```diff
@@ -0,0 +1,109 @@
+- if the product is the main variant, at least one of her variants must be visible
+
+**Offered** - products that satisfied the conditions for **visible** and at the same time they appear in the database with the attribute `calculatedSellingDenied` = `FALSE`.
+The `calculatedSellingDenied` attribute shows whether the product is already sold out or if the product is a variant with the main variant that is not set up with selling denied.
+
+**Listable** - products that satisfied the conditions for **offered** and at the same time, these products are not the variants.
+By default, the specific variants are not included in the product lists.
+Only the main variants are included in the product lists.
+
+**Sellable** - products that satisfied the conditions for **offered** and at the same time, these products are not the main variants (in other words, these products are either the specific variants or they are the standard products).
+Sellable products are products that can actually be purchased.
+Only the standard products or the specific variants can actually be purchased.
+
+## How calculated attributes work?
+Some attributes that are used on the Shopsys Framework are not set directly, but their value stands out as a result of the values of other attributes.
+The values of these special attributes are calculated automatically.
+For example, if a category of products does not have a name for a locale of the specific domain, this category will be automatically set as invisible on this domain.
+The recalculations of these special attributes can be initialized as `immediate` or `scheduled`:
+
+**immediate** - recalculation is initialized when the event `kernel.response` is caught.
+See a class `Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPriceRecalculator` and a method `onKernelResponse`.
+
+**scheduled** - recalculation is initialized later.
+A product or a category of products can be marked for scheduled recalculation, the recalculation itself is initialized with a cron module, see a class `Shopsys\FrameworkBundle\Command\RecalculationsCommand`.
+
+For example, a method `edit` of a class `Shopsys\FrameworkBundle\Model\Product\ProductFacade` calls a method `scheduleProductForImmediateRecalculation` of a class `Shopsys\FrameworkBundle\Model\Product\Availability\ProductAvailabilityRecalculationScheduler`.
+Through this request, the product is marked for `immediate` recalculation of availability.
+The recalculation itself is initialized when the event `kernel.response` is caught by a method `onKernelResponse`.
+It can be seen from this description that the product does not have recalculated availability immediately after the method `edit` is completed.
+This approach with `kernel.response` event is a legacy feature and it will be removed in the future, see [Recalculating products availability and prices immediately instead of on finish request](https://github.com/shopsys/shopsys/issues/202).
+
+## How do I change the environment?
```

Comment:
I would change it to: How do I switch the environment.

Reply:
Why?

Reply:
I never heard something about "environment switch". And "environment change" is quite common.